### ISSUE TITLE
Wondering how the layout works?

### ISSUE THREAD

**Issue by lmihalkovic** [svelto/svelto#3]
It is not so much an issue as a question about how things layout.

What is the role of that declaration? (and others related to that same gutter/2)? I am asking because I was having some difficulties being able to predict the layout of some simple things, and without it ... things just align properly

```
.multiple > .widget:last-child, .pager > .widget:last-child, .tagbox-tags > .widget:last-child, .multiple > .centerer:last-child, .pager > .centerer:last-child, .tagbox-tags > .centerer:last-child, .multiple > .placeholder:last-child, .pager > .placeholder:last-child, .tagbox-tags > .placeholder:last-child, .multiple > .container:last-child, .pager > .container:last-child, .tagbox-tags > .container:last-child, .multiple > .card-header:last-child:not(.multiple), .pager > .card-header:last-child:not(.multiple), .tagbox-tags > .card-header:last-child:not(.multiple), .multiple > .card-block:last-child:not(.multiple), .pager > .card-block:last-child:not(.multiple), .tagbox-tags > .card-block:last-child:not(.multiple), .multiple > .card-footer:last-child:not(.multiple), .pager > .card-footer:last-child:not(.multiple), .tagbox-tags > .card-footer:last-child:not(.multiple), .multiple > .datepicker:last-child, .pager > .datepicker:last-child, .tagbox-tags > .datepicker:last-child, .multiple > fieldset:last-child, .pager > fieldset:last-child, .tagbox-tags > fieldset:last-child, .multiple > .infobar:last-child, .pager > .infobar:last-child, .tagbox-tags > .infobar:last-child, .multiple > .tagbox:last-child, .pager > .tagbox:last-child, .tagbox-tags > .tagbox:last-child, .multiple > .card:last-child, .pager > .card:last-child, .tagbox-tags > .card:last-child, .multiple > .divider.block:last-child, .pager > .divider.block:last-child, .tagbox-tags > .divider.block:last-child, .multiple > hr.vertical:last-child, .pager > hr.vertical:last-child, .tagbox-tags > hr.vertical:last-child, .multiple > .divider.linear.vertical:last-child:empty, .pager > .divider.linear.vertical:last-child:empty, .tagbox-tags > .divider.linear.vertical:last-child:empty, .multiple > .vertical.divider:last-child:empty:not(.block):not(.text), .pager > .vertical.divider:last-child:empty:not(.block):not(.text), .tagbox-tags > .vertical.divider:last-child:empty:not(.block):not(.text), .multiple > .divider.linear.horizontal:last-child:not(:empty), .pager > .divider.linear.horizontal:last-child:not(:empty), .tagbox-tags > .divider.linear.horizontal:last-child:not(:empty), .multiple > .divider.linear:last-child:not(:empty):not(.vertical), .pager > .divider.linear:last-child:not(:empty):not(.vertical), .tagbox-tags > .divider.linear:last-child:not(:empty):not(.vertical), .multiple > .divider:last-child:not(:empty):not(.vertical):not(.block):not(.text), .pager > .divider:last-child:not(:empty):not(.vertical):not(.block):not(.text), .tagbox-tags > .divider:last-child:not(:empty):not(.vertical):not(.block):not(.text), .multiple > .horizontal.divider:last-child:not(:empty):not(.block):not(.text), .pager > .horizontal.divider:last-child:not(:empty):not(.block):not(.text), .tagbox-tags > .horizontal.divider:last-child:not(:empty):not(.block):not(.text), .multiple > .divider.linear.vertical:last-child, .pager > .divider.linear.vertical:last-child, .tagbox-tags > .divider.linear.vertical:last-child, .multiple > .vertical.divider:last-child:not(.block):not(.text), .pager > .vertical.divider:last-child:not(.block):not(.text), .tagbox-tags > .vertical.divider:last-child:not(.block):not(.text), .multiple > .divider.text:last-child, .pager > .divider.text:last-child, .tagbox-tags > .divider.text:last-child, .multiple > .embed:last-child, .pager > .embed:last-child, .tagbox-tags > .embed:last-child, .multiple > form:last-child, .pager > form:last-child, .tagbox-tags > form:last-child, .multiple > label:last-child, .pager > label:last-child, .tagbox-tags > label:last-child, .multiple > header:last-child, .pager > header:last-child, .tagbox-tags > header:last-child, .multiple > footer:last-child, .pager > footer:last-child, .tagbox-tags > footer:last-child, .multiple > i.icon:last-child, .pager > i.icon:last-child, .tagbox-tags > i.icon:last-child, .multiple > .icons:last-child, .pager > .icons:last-child, .tagbox-tags > .icons:last-child, .multiple > img:last-child, .pager > img:last-child, .tagbox-tags > img:last-child, .multiple > input:last-child, .pager > input:last-child, .tagbox-tags > input:last-child, .multiple > textarea:last-child, .pager > textarea:last-child, .tagbox-tags > textarea:last-child, .multiple > .label:last-child, .pager > .label:last-child, .tagbox-tags > .label:last-child, .multiple > .button:last-child, .pager > .button:last-child, .tagbox-tags > .button:last-child, .multiple > .sections:last-child, .pager > .sections:last-child, .tagbox-tags > .sections:last-child, .multiple > .section-left:last-child, .pager > .section-left:last-child, .tagbox-tags > .section-left:last-child, .multiple > .section-center:last-child, .pager > .section-center:last-child, .tagbox-tags > .section-center:last-child, .multiple > .section-right:last-child, .pager > .section-right:last-child, .tagbox-tags > .section-right:last-child, .multiple > .select:last-child, .pager > .select:last-child, .tagbox-tags > .select:last-child, .multiple > .circle:last-child, .pager > .circle:last-child, .tagbox-tags > .circle:last-child, .multiple > .square-sample:last-child, .pager > .square-sample:last-child, .tagbox-tags > .square-sample:last-child, .multiple > .square:last-child, .pager > .square:last-child, .tagbox-tags > .square:last-child, .multiple > .triangle:last-child, .pager > .triangle:last-child, .tagbox-tags > .triangle:last-child, .multiple > .spacer:last-child, .pager > .spacer:last-child, .tagbox-tags > .spacer:last-child, .multiple > svg.spinner:last-child, .pager > svg.spinner:last-child, .tagbox-tags > svg.spinner:last-child, .multiple > .spinner-label:last-child, .pager > .spinner-label:last-child, .tagbox-tags > .spinner-label:last-child, .multiple > .table:last-child, .pager > .table:last-child, .tagbox-tags > .table:last-child, .multiple > .toolbar:last-child, .pager > .toolbar:last-child, .tagbox-tags > .toolbar:last-child, .multiple > address:last-child, .pager > address:last-child, .tagbox-tags > address:last-child, .multiple > blockquote:last-child, .pager > blockquote:last-child, .tagbox-tags > blockquote:last-child, blockquote.multiple > footer:last-child, blockquote.pager > footer:last-child, blockquote.tagbox-tags > footer:last-child, .multiple > h1:last-child, .pager > h1:last-child, .tagbox-tags > h1:last-child, .multiple > h2:last-child, .pager > h2:last-child, .tagbox-tags > h2:last-child, .multiple > h3:last-child, .pager > h3:last-child, .tagbox-tags > h3:last-child, .multiple > h4:last-child, .pager > h4:last-child, .tagbox-tags > h4:last-child, .multiple > h5:last-child, .pager > h5:last-child, .tagbox-tags > h5:last-child, .multiple > h6:last-child, .pager > h6:last-child, .tagbox-tags > h6:last-child, .multiple > ul:last-child, .pager > ul:last-child, .tagbox-tags > ul:last-child, .multiple > ol:last-child, .pager > ol:last-child, .tagbox-tags > ol:last-child, .multiple > dl:last-child, .pager > dl:last-child, .tagbox-tags > dl:last-child, .multiple > p:last-child, .pager > p:last-child, .tagbox-tags > p:last-child, .multiple > pre:last-child, .pager > pre:last-child, .tagbox-tags > pre:last-child 
{
    margin-bottom: 0.625rem;
}
```

**Comment by fabiospampinato** [comment]
I'm closing this as a duplicate of #2 as really the problem here is the missing documentation.

Trying to answer your question:

1. First of all you'd have to know of SCSS works, that selector is automatically generated.
2. There's a "[widget](https://github.com/svelto/svelto/blob/98989046abec54e24d09a48d8a7e43955f69264f/src/core/widget/widget.scss#L7)" primitive, which has a bottom margin equal to `$gutter`, so that if we use multiple widgets they stack up nicely
3. There's a "[multiple](https://github.com/svelto/svelto/blob/98989046abec54e24d09a48d8a7e43955f69264f/src/core/multiple/multiple.scss)" primitive, which is used for grouping multiple things together. For instance other frameworks such as Bootstrap have things like [`.btn-group`](https://getbootstrap.com/docs/4.1/components/button-group/), for grouping multiple buttons together, instead we have `.multiple` for grouping _all kinds_ of things.
4. The snippet to quoted comes from [here](https://github.com/svelto/svelto/blob/98989046abec54e24d09a48d8a7e43955f69264f/src/core/widget/widget.scss#L5). The point of that is this: the last widget inside a widget is separated at the top from the previous widget because the previous widget has a margin-bottom, but at the bottom the last widget doesn't have to separate itself from anything, as there are no other widgets after it, and its parent widget is already providing the margin for other widgets that are 1 level higher in the hierarchy.

Basically this is the result:

<img width="575" alt="screen shot 2019-02-17 at 18 02 14" src="https://user-images.githubusercontent.com/1812093/52916330-4697e900-32de-11e9-87f9-c8f699824bde.png">

And this is what we get without that:

<img width="571" alt="screen shot 2019-02-17 at 18 02 34" src="https://user-images.githubusercontent.com/1812093/52916331-4697e900-32de-11e9-9002-3a60d6ea9d9f.png">

**Comment by lmihalkovic** [comment]
Yes, it is a matter of doc. Thank you for your explanation. I have been around long enought to work with all the css generators. 

You might recall the image i shared somewhere else. On it, nothing could make the buttons align with the center of the container they sat inside of (I placed the sort controls with with the del/pin/fav buttons). I tried many different routes and could never get things pixel perfect. And this was also altering the look of the couple of ‘multiple vertical’ blocks i put in the sidebar (the bottom marging was preventing them from aligning perfectly). 
Then i removed both the margin bottom (from all the widgets and from the ‘last-child’), and everything is now perfectly aligned. So i was really pondering under what circumstances these settings were required.

Thank you.

**Comment by lmihalkovic** [comment]
Ok ... I think I understand .. the widget declaration is for encoding this special separator behavior... so i have to ensure that the ‘widget’ behavior is not there... 

But somehow i think (no proof) that a declaration might be missing to avoid doing this bottom margin when the is a ‘multiple’ with a ‘horizontal’ .. because that is also altering how the buttons line up on the toolbats inside Notable.. but as i said, just a hunch which i have not yet tried to verify.
